src/verify.c

```diff
@@ -108,11 +117,44 @@ static void print_help(const char *name)
 	fprintf(stderr, "\n");
 }
 
+static bool compile_whitelist(const char *regex)
+{
+	int errcode;
+
+	if (picky_whitelist) {
+		regfree(picky_whitelist);
+	} else {
+		picky_whitelist = calloc(1, sizeof(*picky_whitelist));
+		if (!picky_whitelist) {
+			fprintf(stderr, "out of memory\n");
+			return false;
+		}
+	}
+
+	errcode = regcomp(picky_whitelist, regex, REG_NOSUB|REG_EXTENDED);
+	if (errcode) {
+		char *buffer;
+		size_t len;
+		len = regerror(errcode, picky_whitelist, NULL, 0);
+		buffer = malloc(len);
+		if (!buffer) {
+			fprintf(stderr, "out of memory\n");
+			return false;
+		}
+		regerror(errcode, picky_whitelist, buffer, len);
+		fprintf(stderr, "Invalid --picky-whitelist=%s: %s\n", regex, buffer);
+		return false;
```

Comment:
Will fix.

Reply:
Regarding ``picky_whitelist = NULL``: the documentation does not explicitly say whether regfree() can or must be called after a failed regcomp(). My interpretation was that it needs to be called because regcomp() might have stored dynamically allocated error information in the regex_t for regerror(). Therefore not unsetting picky_whitelist is intentionally.